apps/examples/testcase/le_tc/kernel/tc_sched.c

```diff
@@ -61,6 +62,39 @@ static void sched_foreach_callback(struct tcb_s *tcb, void *arg)
 }
 
 /**
+* @fn                   :function_task_setcanceltype
+* @description          :Function for tc_sched_task_setcanceltype
+* @return               :int
+*/
+#ifdef CONFIG_CANCELLATION_POINTS
+static int function_task_setcanceltype(int argc, char *argv[])
+{
+	int ret_chk;
+	int type;
+	int oldtype;
+	struct tcb_s *stcb = this_task();
+
+	/* Negative case with invalid mode. It will return EINVAL & set errno flag */
+	type = TASK_CANCEL_INVALID;
+	ret_chk = task_setcanceltype(type, &oldtype);
+	TC_ASSERT_EQ_RETURN("task_setcanceltype", ret_chk, EINVAL, 0);
```

Comment:
When it fails, this function (this task) will be ended with increasing failure number. But parents task (means tc_sched_task_setcanceltype function) does not know this result. So it will execute TC_SUCCESS_RESULT. It is not a right operation. If you want to make a new task to execute, you need to concern this case.